### PR TITLE
add geo support

### LINKED ISSUE #3: Add get user location & sort the data from the nearest location
## Description

Get user location from the browser, and then calculate the distance from the user's location to every available hospital.

## Implementation Model

I think it's more easy to implement at the list page first, and then after that we can replicate the behavior to the map page. We can use `haversine` formula to calculate the distance.

```js
function deg2rad(deg: number) {
  return deg * (Math.PI / 180);
}

export function getDistanceFromLatLonInKm(lat1: number, lon1: number, lat2: number, lon2: number) {
  const R = 6371; // Radius of the earth in km
  const dLat = deg2rad(lat2 - lat1); // deg2rad below
  const dLon = deg2rad(lon2 - lon1);
  const a =
    Math.sin(dLat / 2) * Math.sin(dLat / 2) +
    Math.cos(deg2rad(lat1)) * Math.cos(deg2rad(lat2)) * Math.sin(dLon / 2) * Math.sin(dLon / 2);
  const c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1 - a));
  const d = R * c; // Distance in km
  return d.toFixed(2);
}
```

## Tasks

- [x] Add button to get current location
- [x] Clicking the button will get the nearest province
- [x] Redirect to `/[province]` page with geo info
- [x] Update data to sort by geo info if any

### PR DESCRIPTION
**Pull request by pveyes**
resolve #3 

- move `provincesWithCities` from `pages/api/bed` to `utils/constants`
- add `lat` & `lon` to `provinceWithCities`
- add location button to search with current location, convert to `geo` query string to `[province]` page 
- update `useHospitalDataByProvice` to sort by geo if any

**Preview**

https://user-images.githubusercontent.com/1614415/124852368-6b37ad00-dfce-11eb-95ea-6f72637b01da.mp4

### CODE REVIEW

**Comment by pveyes** [comment]
This PR doesn't change the UI much for future iteration

Things that can be added
 - Display distance data in card
 - Show other nearest province (in case wrongly attributed, e.g: for people in _perbatasan_)

File: hooks/useHospitalDataByProvince.js [hunk 1]
```diff
@@ -13,6 +14,29 @@ export default function useHospitalDataByProvince(province) {
...
+          return {
+            ...hospital,
+            distance,
+          }
+        })
+        .sort((a, b) => {
+          return a.distance > b.distance ? 1 : -1
+        })
+
+      console.log(hospitalList)
```
Comment (masbagal) [inline hooks/useHospitalDataByProvince.js]: console.log 😛

Comment (pveyes) [reply hooks/useHospitalDataByProvince.js]: hahahah kebiasaan

**Review by masbagal — APPROVED** [review]

**Review by agallio — COMMENTED** [review]
Super! LGTM. Let's 🚢 it.

**Review by agallio — APPROVED** [review]

**Pull request merged**